### ISSUE TITLE
[Usage]  train llava-v1.5-13b-pretrain  "loss":0..0

### ISSUE THREAD

**Issue by wanlipeng** [haotian-liu/LLaVA#1621]

**Comment by wanlipeng** [comment]
[2024-07-21 09:55:28,417] [INFO] [real_accelerator.py:161:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-07-21 09:55:30,617] [INFO] [launch.py:138:main] 0 NCCL_VERSION=2.11.4
[2024-07-21 09:55:30,617] [INFO] [launch.py:145:main] WORLD INFO DICT: {'localhost': [0]}
[2024-07-21 09:55:30,617] [INFO] [launch.py:151:main] nnodes=1, num_local_procs=1, node_rank=0
[2024-07-21 09:55:30,617] [INFO] [launch.py:162:main] global_rank_mapping=defaultdict(<class 'list'>, {'localhost': [0]})
[2024-07-21 09:55:30,617] [INFO] [launch.py:163:main] dist_world_size=1
[2024-07-21 09:55:30,617] [INFO] [launch.py:165:main] Setting CUDA_VISIBLE_DEVICES=0
[2024-07-21 09:55:33,036] [INFO] [real_accelerator.py:161:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-07-21 09:55:37,824] [INFO] [comm.py:637:init_distributed] cdb=None
[2024-07-21 09:55:37,824] [INFO] [comm.py:668:init_distributed] Initializing TorchBackend in DeepSpeed with backend nccl

You are using a model of type llama to instantiate a model of type llava_llama. This is not supported for all configurations of models and can yield errors.
You are attempting to use Flash Attention 2.0 with a model not initialized on GPU. Make sure to move the model to GPU after initializing it on CPU with `model.to('cuda')`.
Loading checkpoint shards:   0%|                                                                                                                      | 0/3 [00:00<?, ?it/s]/opt/conda/envs/llava/lib/python3.10/site-packages/torch/_utils.py:776: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
  return self.fget.__get__(instance, owner)()

/opt/conda/envs/llava/lib/python3.10/site-packages/transformers/generation/configuration_utils.py:392: UserWarning: `do_sample` is set to `False`. However, `temperature` is set to `0.9` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `temperature`. This was detected when initializing the generation config instance, which means the corresponding file may hold incorrect parameterization and should be fixed.
  warnings.warn(
/opt/conda/envs/llava/lib/python3.10/site-packages/transformers/generation/configuration_utils.py:397: UserWarning: `do_sample` is set to `False`. However, `top_p` is set to `0.6` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `top_p`. This was detected when initializing the generation config instance, which means the corresponding file may hold incorrect parameterization and should be fixed.
  warnings.warn(
Formatting inputs...Skip in lazy mode

/opt/conda/envs/llava/lib/python3.10/site-packages/torch/utils/checkpoint.py:31: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
{'loss': 9.3847, 'learning_rate': 1.4326647564469915e-06, 'epoch': 0.0}                                                                                                     
{'loss': 8.7351, 'learning_rate': 2.865329512893983e-06, 'epoch': 0.0}                                                                                                      
{'loss': 0.0, 'learning_rate': 4.2979942693409744e-06, 'epoch': 0.0}                                                                                                        
{'loss': 0.0, 'learning_rate': 5.730659025787966e-06, 'epoch': 0.0}                                                                                                         
{'loss': 0.0, 'learning_rate': 7.163323782234957e-06, 'epoch': 0.0}                                                                                                         
{'loss': 0.0, 'learning_rate': 8.595988538681949e-06, 'epoch': 0.0}                                                                                                         
{'loss': 0.0, 'learning_rate': 1.0028653295128941e-05, 'epoch': 0.0}                                                                                                        
{'loss': 0.0, 'learning_rate': 1.1461318051575932e-05, 'epoch': 0.0}                                                                                                        
{'loss': 0.0, 'learning_rate': 1.2893982808022922e-05, 'epoch': 0.0}                                                                                                        
{'loss': 0.0, 'learning_rate': 1.4326647564469915e-05, 'epoch': 0.0}                                                                                                        
{'loss': 0.0, 'learning_rate': 1.5759312320916907e-05, 'epoch': 0.0}                                                                                                        
{'loss': 0.0, 'learning_rate': 1.7191977077363898e-05, 'epoch': 0.0}                                                                                                        
{'loss': 0.0, 'learning_rate': 1.862464183381089e-05, 'epoch': 0.0}                                                                                                         
{'loss': 0.0, 'learning_rate': 2.0057306590257882e-05, 'epoch': 0.0}                                                                                                        
{'loss': 0.0, 'learning_rate': 2.148997134670487e-05, 'epoch': 0.0}

**Comment by WayneTomas** [comment]
the same proble

**Comment by wanlipeng** [comment]
@WayneTomas I tried to downgrade the deepspeed library, and the loss seems normal now. I'm not sure what caused it, anyone else know the this problem?
-----------------------------------------------
deepspeed    0.9.5
pydantic        1.10.7  
torch              2.0.1
torchvision     0.16.2
flash_attn-2.1.1

**Comment by MoyusiteruIori** [comment]
have the same problem, but downgrading deepspeed does not work for me😭

**Comment by WayneTomas** [comment]
> wanlipeng

I keep the same version of these packages as wanlipeng's, and it works.

**Comment by WayneTomas** [comment]
> have the same problem, but downgrading deepspeed does not work for me😭

I keep the same version of these packages as wanlipeng's, and it works.

**Comment by MoyusiteruIori** [comment]
> > have the same problem, but downgrading deepspeed does not work for me😭
> 
> I keep the same version of these packages as wanlipeng's, and it works.

OK, my problem was because some wrong arguments in the training script. It turns out that the code works well for me without downgrading any package.😵‍💫Thank you anyway for the reply!
